### DISCUSSION TITLE
How I can enable spell checking only for comments and text used in String varaibles?

### DISCUSSION THREAD

**Discussion by tarekahf** [streetsidesoftware/vscode-spell-checker#1558] [General]
This is working just fine. However, I am finding it a little annoying that there are lots of spelling mistakes.

- How I can turn off spell check for everything except comments for example?
- How I can turn off spell check for everything except comments and text in a String variable?

I appreciate your help.

Tarek

**Comment by jnz86** [comment]
For C example, in the settings.json. A pre-defined comment regex which exists for at least C, and one I made that could definitely be wrong but seems to work.
```
    "cSpell.languageSettings": [
        {
            "languageId": "C",
            "includeRegExpList": [
                "CStyleComment",
                "\"[^\"]*\""
            ]
        }
    ]
```

I don't know why this was left as an open issue for so long. I found this post before I found the real answer. That it can be done, but only per language.
https://github.com/streetsidesoftware/vscode-spell-checker/issues/107

**Comment by ErezArbell** [comment]
This is a needed feature: to check spelling only in comments and/or strings. Globally or per language.

**Comment by Jason3S** [comment]
@ErezArbell,

Work is underway to support feature. But funding is low.

Relate to: [Parsing files](https://github.com/streetsidesoftware/cspell/tree/main/rfc/rfc-0003%20parsing%20files#readme)

**Reply by tarekahf** [reply]
How to support the funding?

**Reply by Jason3S** [reply]
[CSpell - Open Collective](https://opencollective.com/cspell)

or

- Become a [<img src="https://github.githubassets.com/images/modules/site/icons/funding_platforms/patreon.svg" width="16" height="16" alt="Patreon">Patreon!](https://patreon.com/streetsidesoftware)
- [Support through ![PayPal](https://raw.githubusercontent.com/streetsidesoftware/vscode-spell-checker/main/images/PayPal/paypal-logo-wide-18.png)](https://www.paypal.com/donate/?hosted_button_id=26LNBP2Q6MKCY)
